src/datachain/nodes_thread_pool.py

```diff
@@ -57,44 +57,65 @@ def __init__(self, max_threads):
         self._max_threads = max_threads
         self._thread_counter = 0
         self._thread_lock = threading.Lock()
```

Comment:
separate: probably can be done w/o a lock if this only about `+1`. locks imposes some performance loss. There should be something like atomicint or something